### ISSUE TITLE
Support for !case statement in macros.

### ISSUE THREAD

**Issue by Aquaricat** [UnicornHeartClub/Issues#60]
Currently, a missing capability in TTML is the ability to write a select case analog.  While this is a bit more 'programmy', it also would simplify writing certain kinds of macros, such as working with level-based features.  For example:
* Warlock Spell Casting
  * Warlocks always cast their spells at the highest level, so a case that checks if warlock level is > x and <= y would simplify setting up spell casting macros for them.
* Cantrip Die Count
  * Cantrips scale based on character level, so a simple case could be setup to check.
* Monk Martial Arts Movement
  * Monks gain bonuses to their speed when unarmored and unshielded, which scales depending on armor.  In this case, a check could be made to see if the PC is unarmored/unshielded, and then call a sub-macro that returns the appropriate speed bonus.
* Monk Martial Arts Damage Die size
  * Monks can replace the damage of certain weapons with a different die based on their level.

Functionally, it'd work fairly similarly to `!prompt`, it's just that the value is being supplied in advance.

**Comment by Aquaricat** [comment]
A few additional thoughts:
* The test portion of each case does need to support multiple conditionals for optimal functionality, `$var > 4 && $var <= 7` for example, as the key use for this is working with ranges.
* Question: Can TTML handle being passed a full die roll as a `$var`?  IE,
```
$die = 1d4
/r $die
```

**Comment by Pjb518** [comment]
At the moment, I think we have to use `1d4 >> $1` which stores the result of the roll, not the roll setup. I just ran this example and got a type error:
```bash
$var = 1d4
!roll $var
```

**Comment by Pjb518** [comment]
> The test portion of each case does need to support multiple conditionals for optimal functionality, `$var > 4 && $var <= 7` for example, as the key use for this is working with ranges.

My personal feeling regarding things like && and || is that we should avoid them in favour of key words, because it makes the language less scary to somebody without programming experience. I believe we should restrict strings to quoted text and reserve a number of words like default, and, or, etc. for special cases. The example above would then look like this: `$var > 4 and $var <= 7`

**Comment by Aquaricat** [comment]
On the one hand, I agree that there should be keyword equivalents.  On the other, IMHO this would be a syntactic sugar - it could easily support both.

**Comment by Pjb518** [comment]
I have no issues with supporting both. I guess the other side of the coin is that people _with_ programming experience are going to expect && and ||, so perhaps it's best not to make something which throws them off.

**Comment by Aquaricat** [comment]
> I just ran this example and got a type error

This is something that should probably be addressed in a future update.  Being able to store the entire die roll as a $var would be convenient and probably easier to read in this case, such as for the monk damage die.

**Comment by Pjb518** [comment]
Indeed. Definitely not a pressing update though as we can do the following in the meantime:
```bash
$die = 8
!r 1d$die + @me.dex_mod
```
or something along those lines. It is, however, not a particularly easy thing to read, and I can see this becoming a pain in longer macros.

**Comment by Aquaricat** [comment]
Yup.  Consider Toll the Bell:
```
$cantrip = #cantrip_die
$die = #if_enemy_hp_full
!r $cantripd#die
```
...Hrm. I think THAT example just pointed something new out, too.
